api/ttn/lorawan/v3/qrcodegenerator.proto

```diff
@@ -130,3 +131,53 @@ service EndDeviceQRCodeGenerator {
     };
   }
 }
+
+message ParseGatewayQRCodeRequest {
+  // QR code format identifier.
+  // If this field is not specified, the server will default to ttigpro1.
+  string format_id = 1 [(validate.rules).string = {
+    pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$|^$",
+    max_len: 36
+  }];
+  // Raw QR code contents.
+  bytes qr_code = 2 [(validate.rules).bytes = {
+    min_len: 10,
+    max_len: 1024
+  }];
+}
+
+message ParseGatewayQRCodeResponse {
+  // Identifier of the format used to successfully parse the QR code data.
+  string format_id = 1;
+  bytes gateway_eui = 2 [
+    (validate.rules).bytes = {
+      len: 8,
+      ignore_empty: true
+    },
+    (thethings.json.field) = {
+      marshaler_func: "go.thethings.network/lorawan-stack/v3/pkg/types.MarshalHEXBytes",
+      unmarshaler_func: "go.thethings.network/lorawan-stack/v3/pkg/types.Unmarshal8Bytes"
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      type: STRING,
+      format: "string",
+      example: "\"70B3D57ED000ABCD\""
+    }
+  ];
+  string owner_token = 3 [(validate.rules).string.pattern = "^[A-Z0-9]{1,32}$"];
```

Comment:
Remove validation here. This is a response so there's no validation required. It could communicate what contents `owner_token` might have, but we don't know that at the API level because it depends on the (future) QR code formats

Reply:
done